pkgs/development/compilers/haxe/default.nix

```diff
@@ -1,14 +1,39 @@
-{ lib, stdenv, fetchgit, coreutils, ocamlPackages, zlib, pcre, neko }:
-
-let inherit (ocamlPackages) ocaml camlp4; in
+{ lib, stdenv, fetchgit, coreutils, ocaml-ng, zlib, pcre, neko, mbedtls }:
 
 let
-  generic = { version, sha256, prePatch }:
+  ocamlDependencies = version:
+    if lib.versionAtLeast version "4.0"
+    then with ocaml-ng.ocamlPackages; [
+      ocaml
+      findlib
+      sedlex_2
+      xml-light
+      xml-light
+      ptmap
+      camlp5
+      sha
+      dune_2
+      luv
+    ] ++ (if lib.versionAtLeast version "4.2" then [
+      ocaml_extlib_1_7_8
+    ] else [
+      ocaml_extlib
+    ])
+    else with ocaml-ng.ocamlPackages_4_05; [
+      ocaml
+      camlp4
+    ];
+
+  # extraPatch gets appended to the default prePatch. If prePatch is given,
+  # it overrides the default prePatch and extraPatch is irrelevant
+  generic = { sha256, version, extraPatch ? "", ... }@args:
     stdenv.mkDerivation {
       pname = "haxe";
       inherit version;
 
-      buildInputs = [ocaml zlib pcre neko camlp4];
+      buildInputs = [ zlib pcre neko ]
+        ++ lib.optional (lib.versionAtLeast version "4.1") [ mbedtls ]
+        ++ ocamlDependencies version;
 
       src = fetchgit {
```

Comment:
Can we use fetchFromGitHub?

Reply:
I guess, but `fetchFromGitHub` also just calls `fetchgit` in that case.